storm-core/src/jvm/org/apache/storm/daemon/supervisor/SupervisorUtils.java

```diff
@@ -0,0 +1,273 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.daemon.supervisor;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.curator.utils.PathUtils;
+import org.apache.storm.Config;
+import org.apache.storm.generated.LSWorkerHeartbeat;
+import org.apache.storm.localizer.LocalResource;
+import org.apache.storm.localizer.Localizer;
+import org.apache.storm.utils.ConfigUtils;
+import org.apache.storm.utils.LocalState;
+import org.apache.storm.utils.Utils;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.data.ACL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URLDecoder;
+import java.util.*;
+
+public class SupervisorUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SupervisorUtils.class);
+
+    private static final SupervisorUtils INSTANCE = new SupervisorUtils();
+    private static SupervisorUtils _instance = INSTANCE;
+
+    public static void setInstance(SupervisorUtils u) {
+        _instance = u;
+    }
+
+    public static void resetInstance() {
+        _instance = INSTANCE;
+    }
+
+    public static Process workerLauncher(Map conf, String user, List<String> args, Map<String, String> environment, final String logPreFix,
```

Comment:
It does launch the worker when run-as-user is set, that was what it was originally for, but other operations needed privileges too, and as such it has expanded its role.

I am fine with changing it, just thought I would give you some background on it.  FYI we took most of the code from the hadoop `container-launcher`, and they have the same naming problem.

Ideally I would ultimately like to see a  WorkerManager java API that would call into a native linux-worker-manager binary and would look something like the following, NOTE: I have not thought through all of the APIs and I have really not thought about exactly what parameters would need to be passed to it.  This is just a general concept that would allow us to eventually support docker, local-workers, run-as-user, cgroups and possibly other things much more cleanly.

```
public interface WorkerManager {
  Future<WorkerPrepareResult> prepareWorker(String workerId, LocalisedResources lr, Resources resources);
  Future<LaunchWorkerResult> launchWorker(String workerId, WorkerExitCallback, ...);
  Future<KillWorkerResult> killWorker(String workerId);
  Future<CleanupWorkerResult> cleanupWorker(String workerId,...);
  List<WorkerInfo> runningWorkers();
  //Possibly in the  future
  //Future<ResizeWorkerResult> resizeWorker(String workerId, Resources newResources);
}
```